first-commit

MuhammadNomanShoukat opened the pull request:
PR done.

Commit:
first-commit

tayyab-razzaq commented on weather_app/models.py:
```diff
@@ -0,0 +1,41 @@
+
+    """ return back fields to the admin panel """
+    def __str__(self):
+        return r"{0} - {1} - {2}".format(self.min_value,self.max_value,self.avg_value)
```
use [f-string](https://www.python.org/dev/peps/pep-0498/) for string representation.

tayyab-razzaq commented on weather_app/models.py:
```diff
@@ -0,0 +1,41 @@
+from django.db import models
+import uuid
+
+""" Custiom time stamp model """
```
wrong docstring format, see `pydocstyle` for standard python docstring formats.

tayyab-razzaq commented on weather_app/models.py:
```diff
@@ -0,0 +1,41 @@
+from __future__ import unicode_literals
+from django.db import models
+import uuid
```
why an unused import, implement `pylint with pylint-django plugin` and `pycodestyle` for better and standard python code

tayyab-razzaq commented on weather_app/models.py:
```diff
@@ -0,0 +1,41 @@
+    modified_at = models.DateTimeField(auto_now=True)
+    is_active = models.BooleanField(default=True)
+
+    """ using to set class abstract for to create a model """
```
wrong meaning of meaning of model in docstring its leading to misconception of abtract models

tayyab-razzaq commented on weather_app/migrations/0001_initial.py:
```diff
@@ -0,0 +1,53 @@
+# Generated by Django 2.2.4 on 2019-09-13 12:29
```
docstring missing

tayyab-razzaq commented on weather_app/migrations/0001_initial.py:
```diff
@@ -0,0 +1,53 @@
+                ('events', models.CharField(default='', max_length=5)),
+                ('wind_dir_degrees', models.CharField(default='', max_length=5)),
+                ('dew', models.OneToOneField(on_delete=django.db.models.deletion.CASCADE, related_name='+', to='weather_app.WeatherCharacteristic')),
+                ('humidity', models.OneToOneField(on_delete=django.db.models.deletion.CASCADE, related_name='+', to='weather_app.WeatherCharacteristic')),
```
max line length rule breaking

tayyab-razzaq commented on weather_app/management/commands/utils.py:
```diff
@@ -0,0 +1,61 @@
+def value_validation(value):
+
+    if value == "":
+        value = 0.0
```
whats the purpose of else, if both else and if are returning the same value. and for easy if-else we should prefer pythonic way of if-else like the above code will become `return 0.0 if value == '' else value`

tayyab-razzaq commented on weather_app/management/commands/utils.py:
```diff
@@ -0,0 +1,61 @@
+        return value
+    else:
+        return value
+
```
too many blank lines at the end of file, this breaks python file rule, see pep8 conventions

tayyab-razzaq commented on weather_app/management/commands/utils.py:
```diff
@@ -0,0 +1,61 @@
+    """
+    for root, directory, files in os.walk(path):
+        """ here finding .txt entention files """
+        for file in files:
```
`file` is keyword in many language so not a good approach to use it as a variable name

tayyab-razzaq commented on weather_app/management/commands/utils.py:
```diff
@@ -0,0 +1,61 @@
+import os
+import csv
+from weather_app.models import WeatherCharacteristic,Weather,CustomTimeStampModel
```
import order issue, see pylint and isort for this

tayyab-razzaq commented on weather_app/management/commands/utils.py:
```diff
@@ -0,0 +1,61 @@
+""" reading a files form directory"""
+def reading_files(path):
+    all_files = []
+    """
```
wrong placement of docstrings

tayyab-razzaq commented on weather_app/management/commands/utils.py:
```diff
@@ -0,0 +1,61 @@
+    loop getting files from path
+    sending files to the "all_files" list
+    """
+    for root, directory, files in os.walk(path):
```
would be better if we use `glob` library instead of walking through os, it would be more optimized

tayyab-razzaq commented on weather_app/management/commands/utils.py:
```diff
@@ -0,0 +1,61 @@
+        for file in files:
+            if '.txt' in file:
+                all_files.append(os.path.join(root, file))
+    """ returning the all_files list """
```
for inline comments use comments docstrings are used for different purpose.

tayyab-razzaq commented on weather_app/management/commands/utils.py:
```diff
@@ -0,0 +1,61 @@
+""" sending data to the database fily by bile """
+def data_sending_to_database(files):
+
+    for f in list(files):
```
meaningless variable name `f`

tayyab-razzaq commented on weather_app/management/commands/utils.py:
```diff
@@ -0,0 +1,61 @@
+
+    for f in list(files):
+        with open(f) as csv_file:
+            csv_reader = csv.reader(csv_file, delimiter=',')
```
see `DictReader` of `csv` library that will be more optimized for this scenario.

tayyab-razzaq replied:
default deimiter for csv is `,` no need to specify it.

tayyab-razzaq commented on weather_app/management/commands/utils.py:
```diff
@@ -0,0 +1,61 @@
+            csv_headings = next(csv_reader)
+
+            for row in csv_reader:
+                wc = WeatherCharacteristic(max_value=value_validation(row[1]),
```
on what basis you are choosing row[1], row[7] and row[3]. explain it

tayyab-razzaq commented on weather_app/management/commands/utils.py:
```diff
@@ -0,0 +1,61 @@
+                                           avg_value=value_validation(row[3]))
+                wc.save()
+                w = Weather(weather_date=row[0],
+                            temperature=wc,
```
why all attributes like `temperature`, `humidity`, `dew`, `sea_level_pressure`, `visibility`, `wind_velocity` are saving the same value ?

tayyab-razzaq commented on weather_app/management/commands/utils.py:
```diff
@@ -0,0 +1,61 @@
+                            sea_level_pressure=wc,
+                            visibility=wc, wind_velocity=wc,
+                            precipitation=value_validation(row[19]),
+                            cloud_cover=value_validation(row[20]),
```
if any file doesn't follow this order of headers would your code still works ?, why doing indexing ?

tayyab-razzaq commented on weather_app/management/commands/utils.py:
```diff
@@ -0,0 +1,61 @@
+                            cloud_cover=value_validation(row[20]),
+                            events=value_validation(row[21]),
+                            wind_dir_degrees=value_validation(row[22]))
+                w.save()
```
meaningless variable name `w`

tayyab-razzaq replied:
too many queries while saving the weather instance, as its the parent model bulkCreate would be better.

tayyab-razzaq commented on weather_app/management/commands/load_files_data.py:
```diff
@@ -0,0 +1,23 @@
+    both function are using in it
+    """
+    def handle(self, *args, **options):
+        path_root = os.path.abspath(os.path.dirname(__name__))
```
why using the abspath, we can define path from the user how runs the program, see expanduser method of os library

tayyab-razzaq commented on weather_app/management/commands/load_files_data.py:
```diff
@@ -0,0 +1,23 @@
+from django.core.management.base import BaseCommand
```
wrong import error use pylint to identify and isort for correcting the order of imports

tayyab-razzaq commented on weather_app/management/commands/load_files_data.py:
```diff
@@ -0,0 +1,23 @@
+    """
+    def handle(self, *args, **options):
+        path_root = os.path.abspath(os.path.dirname(__name__))
+        folder_name = "weatherfiles"
```
folder name should be prompted from user, anyone can save files in folder named of their own choice

tayyab-razzaq commented on weather_app/management/commands/load_files_data.py:
```diff
@@ -0,0 +1,23 @@
+        path = path_root + "\\" + folder_name
+        """ using for reading files form directory and returning back to files """
+        files = reading_files(path)
+        """ using for sending data to the database """
```
no need to describe every line, just add comments for complex code.

tayyab-razzaq commented on weather_app/admin.py:
```diff
@@ -0,0 +1,27 @@
+    ]
+
+class WeatherUserAdmin(admin.ModelAdmin):
+    list_display = [
```
I asked you to add custom filters which are missing here

tayyab-razzaq commented on weather_app/admin.py:
```diff
@@ -0,0 +1,27 @@
+        'avg_value',
+    ]
+
+    list_filter = [
```
no need to add such a huge value column for filtering its baseless, try adding a custom filter for ranges of temperature

tayyab-razzaq commented on django_weatherman/settings.py:
```diff
@@ -0,0 +1,121 @@
+# Static files (CSS, JavaScript, Images)
+# https://docs.djangoproject.com/en/2.2/howto/static-files/
+
+STATIC_URL = '/static/'
```
log configuration is still missing, I already informed you to added logging in your project, see django logging module

tayyab-razzaq commented:
Please use meaning full PR names and commit messages and follow a standard structure for commit messages.  see [this](https://udacity.github.io/git-styleguide/) for commit style guide.

tayyab-razzaq commented:
see point-7 in readme for PR name standards.

MuhammadNomanShoukat committed:
Update models.py

Committed files, but not some changes because of files. Some fields exist on Strings like ("RAIN THUNDERSTORM").

MuhammadNomanShoukat committed:
Committed done.

I've tried to make a custom filter, but I don't understand how to make it. I'm trying to make it as soon as possible. There are some piece of code on the internet but I don't know how to handle them. You can se in the commented code.

MuhammadNomanShoukat committed:
Committed done.

In that file, I have set the user base directory, he can make his own directory and make a file in it. but some changes have not done. I have left it to make them better in further process.

MuhammadNomanShoukat committed:
Committed done.

Tayyab bhai I have no idea how to make a parent bulk query creations, So that's why I left them some fields in the Weather module. I tried my best as much as but some things which I don't change I want to make them better in further.

MuhammadNomanShoukat committed:
Committed done.

tayyab-razzaq commented on django_weatherman/settings.py:
```diff
@@ -0,0 +1,143 @@
+        'file': {
+            'level': 'DEBUG',
+            'class': 'logging.FileHandler',
+            'filename': 'weather_app/management/commands/weather_log.log',
```
create a separate folder named logs and add logs files there, and add that folder to gitignore

tayyab-razzaq commented on django_weatherman/settings.py:
```diff
@@ -0,0 +1,143 @@
+    'handlers': {
+        'file': {
+            'level': 'DEBUG',
+            'class': 'logging.FileHandler',
```
add file rotation to log file

tayyab-razzaq commented on weather_app/admin.py:
```diff
@@ -0,0 +1,75 @@
+from .models import Weather, WeatherCharacteristic
+
+
+# class SpeciesListFilter(admin.SimpleListFilter):
```
why this commented code ?

tayyab-razzaq commented on weather_app/admin.py:
```diff
@@ -0,0 +1,75 @@
+from django.contrib import admin
+from django.contrib.admin import SimpleListFilter
```
unused import ? pylint, pycodestyle is still missing

tayyab-razzaq commented on weather_app/admin.py:
```diff
@@ -0,0 +1,75 @@
+from django.contrib import admin
```
module docstring missing ? pydocstyle ?

tayyab-razzaq commented on weather_app/admin.py:
```diff
@@ -0,0 +1,75 @@
+    ]
+
+    list_filter = [
+        'weather_date',
```
it will still become a too much long filter, so we should avoid it, you can use `date_hierarchy` field to filter queryset

tayyab-razzaq commented on weather_app/admin.py:
```diff
@@ -0,0 +1,75 @@
+    ]
+
+
+admin.site.register(WeatherCharacteristic,WeatherCharacteristicAdminClass)
```
pep8 standards are not being followed

tayyab-razzaq commented on weather_app/management/commands/load_files_data.py:
```diff
@@ -0,0 +1,32 @@
+class Command(BaseCommand):
+    help = "Loading data from files"
+
+    """ using for weather file argument '"""
```
docstrings isn't according to standards yet

tayyab-razzaq commented on weather_app/management/commands/load_files_data.py:
```diff
@@ -0,0 +1,32 @@
+
+        """ checking here if directory is already exist or not"""
+        if not os.path.isdir(folder_name):
+            os.mkdir(os.path.join(path_root, folder_name))
```
should log error here that given path is not directory

tayyab-razzaq commented on weather_app/management/commands/load_files_data.py:
```diff
@@ -0,0 +1,32 @@
+        else:
+            path = path_root + "\\" + folder_name
+            files = reading_files(path)
+            data_sending_to_database(files)
```
save_data_to_db sounds good as compared to data_sending_to_database

tayyab-razzaq commented on weather_app/management/commands/utils.py:
```diff
@@ -0,0 +1,86 @@
+    loop getting files from path
+    sending files to the "all_files" list
+    """
+    for root, directory, weather_files in os.walk(path):
```
use `glob` to iterate files in a folder

tayyab-razzaq commented on weather_app/models.py:
```diff
@@ -0,0 +1,40 @@
+
+
+class CustomTimeStampModel(models.Model):
+    created_at = models.DateTimeField(auto_now_add=True)
```
docstrings missing

tayyab-razzaq commented on weather_app/management/commands/utils.py:
```diff
@@ -0,0 +1,86 @@
+
+def value_validation(value):  # This function using for value validation
+
+    if value == "":
```
would be better if written in pythonic way

tayyab-razzaq commented on weather_app/management/commands/utils.py:
```diff
@@ -0,0 +1,86 @@
+                                     cloud_cover=value_validation(row[" CloudCover"]),
+                                     events=value_validation(row[" Events"]),
+                                     wind_dir_degrees=value_validation(row["WindDirDegrees"]))
+            weather.save()
```
store the weather object to list and use `bulkCreate`

tayyab-razzaq commented:
add logging in your project as info as well so that your process progess should be logged as well.
